sdks/java/testing/jpms-tests/build.gradle

```diff
@@ -120,7 +121,7 @@ plugins.withType(JavaPlugin).configureEach{
 // JPMS requires JDK > 8
 project.tasks.each {
   it.onlyIf {
-    project.hasProperty("compileAndRunTestsWithJava17")
+    project.hasProperty('testJavaVersion')
```

Comment:
Looks like this was just running on 17 previously - do we need tighter filtering here to avoid running on 11?

Reply:
Currently Java11 JPMS test is done by setting
 https://github.com/apache/beam/blob/39678b6acd7121a893059236f5d0cb6a61aef9c0/.test-infra/jenkins/job_PostCommit_Java_Jpms_Direct_Java11.groovy#L44
so everything runs on Java11;

while Java17 test is done by setting
https://github.com/apache/beam/blob/39678b6acd7121a893059236f5d0cb6a61aef9c0/.test-infra/jenkins/job_PostCommit_Java_Jpms_Direct_Java17.groovy#L45
so all but compilation of jpms-test project are still run on Java8.

i.e. currently the combination of javahome=java 8+`compileAndRunTestsWithJava11` option does not run test (where it supposed to). This change fixed the issue and make the behavior consistent for Java11/17/21